modules/tinymce/src/core/main/ts/content/PrePostProcess.ts

```diff
@@ -4,13 +4,14 @@ import Editor from '../api/Editor';
 import * as Events from '../api/Events';
 import DomParser from '../api/html/DomParser';
 import HtmlSerializer from '../api/html/Serializer';
+import * as Options from '../api/Options';
 import { EditorEvent } from '../api/util/EventDispatcher';
 import { Content, GetContentArgs, isTreeNode, SetContentArgs } from './ContentTypes';
 
 const serializeContent = (content: Content): string =>
   isTreeNode(content) ? HtmlSerializer({ validate: false }).serialize(content) : content;
 
-const withSerializedContent = <R extends EditorEvent<{ content: string }>>(content: Content, fireEvent: (content: string) => R): R & { content: Content } => {
+const withSerializedContent = <R extends EditorEvent<{ content: string }>>(editor: Editor, content: Content, fireEvent: (content: string) => R): R & { content: Content } => {
```

Comment:
Please don't pass the editor to a method unless it's _really_ needed - keeping the scope of method arguments to a minimum is a technique that can help to reduce accidental dependencies from within the method, aka spaghetti code. It also helps with testing (although I'm not sure that really matters here, just a general point).

In this case, just pass the sanitize option value - it might look like more code to do that, but after minification the difference is negligible.

- [ ] Change method arguments

Reply:
Thanks for the detailed feedback, I've made the change and noted it for the future.